### ISSUE TITLE
Increased Token Limit Not Working with Claude 3.5 Sonnet

### ISSUE THREAD

**Issue by ElectricCodeGuy** [vercel/ai#2298]
### Description

On 2024-07-16, Anthropic announced a new header option to increase the output token limit from 4,000 to 8,192 for Claude 3.5 Sonnet. However, attempts to use this increased limit are resulting in an error.

**Steps to Reproduce:**
1. Set up a request to the Claude 3.5 Sonnet model.
2. Add the header: `'anthropic-beta': 'max-tokens-3-5-sonnet-2024-07-15'`
3. Set `max_tokens` to a value between 4,097 and 8,192 (e.g., 8,192).
4. Execute the API call.

**Expected Behavior:**
The API should accept the request and allow for up to 8,192 output tokens.

**Actual Behavior:**
The API throws an error:
```
AI_APICallError: max_tokens: 8192 > 4096, which is the maximum allowed number of output tokens for claude-3-5-sonnet-20240620
```

[Release information from Anthropic ](https://docs.anthropic.com/en/docs/about-claude/models#model-comparison)

### Code example

```
const { object } = await generateObject({
    model: anthropic('claude-3-5-sonnet-20240620'),
    system: systemPrompt,
    maxTokens: 8192,
    prompt: prompt,
    temperature: 0,
    schema: lawChangeSchema,
    headers: {
      'anthropic-beta': 'max-tokens-3-5-sonnet-2024-07-15'
    }
  });
```

### Additional context

- Model: claude-3-5-sonnet-20240620
- SDK Version Version: [3.2.25](https://github.com/vercel/ai/releases/tag/ai%403.2.25)

**Comment by lgrammel** [comment]
Confirmed that the header is not correctly forwarded.

**Comment by lgrammel** [comment]
Fix: https://github.com/vercel/ai/pull/2299

**Comment by lgrammel** [comment]
Fixed in https://github.com/vercel/ai/releases/tag/ai%403.2.26

**Comment by ElectricCodeGuy** [comment]
That was fast :P 
Good job!!!

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This was a concrete SDK defect in forwarding a request header, causing valid Claude 3.5 Sonnet requests to fail unexpectedly. It was subsequently fixed.

### Evidence
- Requests with the Anthropic beta header still failed at token limits above 4,096.
- A maintainer confirmed that the header was not correctly forwarded.
- The maintainer linked PR #2299 as the fix and stated it was released in AI SDK 3.2.26.
